### PR TITLE
Docker compose example

### PR DESCRIPTION
**Pull request by mclang**
### Purpose

Example about how to spin-up syncthing container using docker compose.

### Testing

Created my current Syncthing instance `docker-compose.yml` like this.

### CODE REVIEW

**Comment by imsodin** [comment]
Can you also expose udp port as done in the docker example since: https://github.com/syncthing/syncthing/pull/7421/files

**Review by pirate — DISMISSED** [review]

**Comment by zilexa** [comment]
FYI with the example from @mclang the config and db are not persistent. They should be as they are not expendable, right?
This is what I use now: 

```
  syncthing:
    container_name: syncthing
    image: syncthing/syncthing
    restart: always
    volumes:
      - ".../docker/syncthing/config:/var/syncthing/config" # persistent config and db
      - ".../user/data/sync:/var/syncthing"
    ports:
      - "8384:8384"
      - "22000:22000/tcp"
      - "22000:22000/udp"
 ```

**Comment by bt90** [comment]
Also relevant: https://github.com/syncthing/syncthing/pull/7536

**Comment by mclang** [comment]
Whoa - now this went completely haywire :astonished: 

I just tried to merge upstream changes into my fork before making any further changes using this as a guide:
https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/syncing-a-fork

Running `git merge upstream/main` worked as per the docs, but merging the changes into my `patch-1` branch gave _cannot fast forward_ error so I tried `rebase` instead of `merge`.

How should I fix this?

**Comment by imsodin** [comment]
Merge should have been enough. You might have first merged and then rebased all the changes on top (i.e. applied diffs again). Just speculating though. Anyways I suggest recreating the branch: Rename the current branch and then create a new branch from main and cherry-pick the relevant commit(s).

**Comment by mclang** [comment]
Some explanation:
After `git merge upstream/main` in my `main` branch, `patch-1` was still of course behind upstream `main` - that is why I did it again in my `patch-1` branch, but it wouldn't fast forward because my one commit, so I tried `rebase` as I have done with my own repositories before.

But okay, so I should now do
```
$ git branch -m patch-1 patch-1-bak
$ git co patch-1
```
But how can I replace the `patch-1` branch on my Github account so that the link to this PR is preserved?

**Comment by mclang** [comment]
Okay, this got closed when I did force push into my branch to clean things up.

Maybe I should have cherry picked my change from the old one before trying if this works :roll_eyes:

**Comment by mclang** [comment]
Now the changes are again present in my branch.

I hope opening this PR picks those up - otherwise I think I need to create new PR.

File: README-Docker.md [hunk 1]
```diff
@@ -7,18 +7,43 @@ Use the `/var/syncthing` volume to have the synchronized files available on the
...
+    environment:
+      - PUID=1000
+      - PGID=1000
+    volumes:
+      - /wherever/st-sync:/var/syncthing
+    ports:
+      - 8384:8384
+      - 22000:22000/tcp
+      - 22000:22000/udp
+    restart: unless-stopped
```
Comment (bt90) [inline README-Docker.md:44]: Could you add add a sysctls section for #7536 ?

https://docs.docker.com/compose/compose-file/compose-file-v3/#sysctls

**Review by bt90 — CHANGES_REQUESTED** [review]

File: README-Docker.md (hunk 1 above)
Comment (mclang) [reply README-Docker.md:44]: The same that is in CLI example above, i.e `net.core.rmem_max=2097152` or the ones in the documentation link?

Comment (bt90) [reply README-Docker.md:46]: I think it's sysctl**s**

**Review by bt90 — CHANGES_REQUESTED** [review]
See my last comment: sysctl**s**

File: README-Docker.md [hunk 2]
```diff
@@ -7,18 +7,45 @@ Use the `/var/syncthing` volume to have the synchronized files available on the
...
+    environment:
+      - PUID=1000
+      - PGID=1000
+    volumes:
+      - /wherever/st-sync:/var/syncthing
+    ports:
+      - 8384:8384
+      - 22000:22000/tcp
+      - 22000:22000/udp
+    sysctl:
```
Comment (bt90) [inline README-Docker.md:44]:
```suggestion
    sysctls:
```

**Review by bt90 — CHANGES_REQUESTED** [review]

File: README-Docker.md (hunk 2 above)
Comment (mclang) [reply README-Docker.md]: Sorry about that :roll_eyes:

**Pull request merged**